From f4be666dcede3a4e988ced43303713659ff68bc3 Mon Sep 17 00:00:00 2001
From: Andrey Gelman <andreyg@compulab.co.il>
Date: Sun, 7 Apr 2013 18:27:27 +0300
Subject: =?UTF-8?q?Workaround=202=20limitations=20of=20SSI=20playback=20(sla?=
 =?UTF-8?q?ve=20mode):=0A-=20severe=20distortions=20playing=20back=20mono=20?=
 =?UTF-8?q?stream=0A-=20severe=20distortions=20playing=20back=208kHz=20sampl?=
 =?UTF-8?q?ing=20rate=20audio?=

The nature of the workaround is SSI reporting it does not
support these modes, making the player convert the audio stream
appropriately.

Reference:
mainline kernel commit
"ASoC: imx-ssi: Remove mono support"
0865a75d4166bddc533fd50831829ceefb94f9b0
---
 sound/soc/imx/imx-ssi.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/imx/imx-ssi.c b/sound/soc/imx/imx-ssi.c
index d398307..a4ff902 100644
--- a/sound/soc/imx/imx-ssi.c
+++ b/sound/soc/imx/imx-ssi.c
@@ -643,15 +643,15 @@ static struct snd_soc_dai_driver imx_ssi_dai = {
 	.suspend = imx_ssi_dai_suspend,
 	.resume = imx_ssi_dai_resume,
 	.playback = {
-		.channels_min = 1,
+		.channels_min = 2,
 		.channels_max = 2,
-		.rates = SNDRV_PCM_RATE_8000_96000,
+		.rates = (SNDRV_PCM_RATE_8000_96000 & ~SNDRV_PCM_RATE_8000),
 		.formats = IMX_SSI_FORMATS,
 	},
 	.capture = {
-		.channels_min = 1,
+		.channels_min = 2,
 		.channels_max = 2,
-		.rates = SNDRV_PCM_RATE_8000_96000,
+		.rates = (SNDRV_PCM_RATE_8000_96000 & ~SNDRV_PCM_RATE_8000),
 		.formats = IMX_SSI_FORMATS,
 	},
 	.ops = &imx_ssi_pcm_dai_ops,
-- 
1.7.9.5

